### PR TITLE
baichi working code

### PR DESCRIPTION
**Pull request by GuaBaiChi**
Hey guys, I tried to add % to the completed part of the !status prompt

### CODE REVIEW

File: src/utils/command_interface.py [hunk 1]
```diff
@@ -159,7 +159,16 @@ def command_status(self, discord_id: str) -> str:
...
             else:
-                result += f"User {leetcode_user_id}'s Weekly Challenge status:\n"
+                questions = self.database.table_weeklyquestion_load_by_challenge_id()
+                total = len(questions)
+                completed = 0
+                leetcode_id = user["leetcode_id"]
+                for q in questions:
+                    if LeetcodeUtil.check_challenge_completion(leetcode_id, q["title_slug"]):
+                        completed += 1
+                percentage = completed/total
```
Comment (jacobmillsfl) [inline src/utils/command_interface.py]: If `total` == 0 then you have a divide by zero error. It's possible that there are no weekly questions for a challenge, or that no challenges exist.

File: src/utils/command_interface.py [hunk 2]
```diff
@@ -159,7 +159,16 @@ def command_status(self, discord_id: str) -> str:
             if not challenge:
                 result += "No current challenges"
             else:
-                result += f"User {leetcode_user_id}'s Weekly Challenge status:\n"
+                questions = self.database.table_weeklyquestion_load_by_challenge_id()
```
Comment (jacobmillsfl) [inline src/utils/command_interface.py]: `table_weeklyquestion_load_by_challenge_id()` loads **by challenge id**. This should take an argument.

File: src/utils/command_interface.py [hunk 3]
```diff
@@ -159,7 +159,16 @@ def command_status(self, discord_id: str) -> str:
...
+                total = len(questions)
+                completed = 0
+                leetcode_id = user["leetcode_id"]
+                for q in questions:
+                    if LeetcodeUtil.check_challenge_completion(leetcode_id, q["title_slug"]):
+                        completed += 1
+                percentage = completed/total
+                result += f"User {leetcode_user_id}'s Weekly Challenge status: {percentage}% \
```
Comment (jacobmillsfl) [inline src/utils/command_interface.py]: See if you can normalize the percentage. The way you have it is going to be some decimal between 0 and 1. We want between 0% and 100% as whole numbers. Do something like `int(percentage * 100)`

Comment (GuaBaiChi) [reply src/utils/command_interface.py]: percentage = int(completed/total) * 100

How about this?

Comment (jacobmillsfl) [reply src/utils/command_interface.py]: No, that is incorrect. `completed` will always be less than or equal to `total`. Therefore, `completed/total` will always be a decimal number between 0.0 and 1.0. If it is anywhere in between, `int(completed/total)` will cast a 0.something decimal into a flat out 0. You need to multiply the result by 100 first, then cast to an int.

**Review by jacobmillsfl — APPROVED** [review]

**Pull request merged**